main.go

```diff
@@ -57,11 +57,13 @@ func validateTools(tools []string) error {
 
 func validatePlan(plan types.Plan) error {
 	for _, secret := range plan.Secrets {
-		if len(secret.Files) > 0 {
-			for _, file := range secret.Files {
-				if len(file.ValueCommand) == 0 {
-					if _, err := os.Stat(file.ExpandValueFrom()); err != nil {
-						return err
+		if featureEnabled(plan.Features, secret.Filters) {
+			if len(secret.Files) > 0 {
```

Comment:
I feel like the complexity / nesting of the function is too high. `secretExists() (bool,error)` ?